### PR TITLE
Create PRs instead of committing to master

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Instead of committing directly to the master branch we now create PRs.
This allows us to enable master branch protection rules.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: .github/workflows/update-tags.yml [hunk 1]
```diff
@@ -14,17 +14,14 @@ jobs:
...
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@2b011faafdcbc9ceb11414d64d0573f37c774b04 # pin@v4.2.3
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          force: false
+          add-paths: gardenctl-v2.rb
+          branch: update_${{ github.event.client_payload.component }}_${{ github.event.client_payload.tag }}
+          delete-branch: true
+          title: update ${{ github.event.client_payload.component }} to ${{ github.event.client_payload.tag }}
+          body: This PR updates gardenctl-v2.rb to the latest version
```
[the remark this answers is not in the document]
Comment (grolu) [reply .github/workflows/update-tags.yml:24]: yes we could do it but it is also already in the title

[the remark this answers is not in the document]
Comment (petersutter) [reply .github/workflows/update-tags.yml:24]: yes, I'd prefer to have it also in the body. E.g. if there are multiple PRs open for different tags saying it's for the `latest version` is misleading ;)

[the remark this answers is not in the document]
Comment (grolu) [reply .github/workflows/update-tags.yml]: done

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**